homeassistant/components/zeroconf/__init__.py

```diff
@@ -259,44 +276,98 @@ async def _async_register_hass_zc_service(
         )
 
 
-async def _async_start_zeroconf_browser(
-    hass: HomeAssistant, zeroconf: Zeroconf
-) -> None:
-    """Start the zeroconf browser."""
+class FlowDispatcher:
+    """Dispatch discovery flows."""
+
+    def __init__(self, hass: HomeAssistant):
```

Comment:
Missing return value typing.

Reply:
I'm surprised mypy didn't complain

Fixed here https://github.com/home-assistant/core/pull/50847

Reply:
Mypy didn't complain because it allows no return type for `__init__`. So it's not typing errors, it's style preference 🙂
Mypy discussion: https://github.com/python/mypy/issues/604

I think we need to enforce this because reviewing it manually every time doesn't scale and it's easy to miss.

@cdce8p can we automate this check using custom pylint plugin?

Reply:
@KapJI In theory you could. Keep in mind however that this would apply to all files not just the type checked once. If you're interested in working on it, I would suggest to open a new Issue (for HomeAssistant) to track this. Alternatively you could also message me on Discord if you need help.